src/commonNative/kotlin/com/epam/drill/agent/Configuration.kt

```diff
@@ -41,6 +42,18 @@ fun performAgentInitialization(initialParams: Map<String, String>) {
     }
 }
 
+private fun calculateBuildVersion(): String? = runCatching {
+    getenv(SYSTEM_JAVA_APP_JAR)?.toKString()?.let {
+        "(.*)/(.*).jar".toRegex().matchEntire(it)?.let { matchResult ->
+            if (matchResult.groupValues.size == 3) {
+                val buildVersion = matchResult.groupValues[2]
+                logger.debug { "calculated build version = '$buildVersion'" }
```

Comment:
a logger is not started yet

Reply:
I don't understand how this regular exp is working.
for example for this jar "recommendation-service-0.0.1.jar" will work this reg (.*)-(.*).jar
And how it will work with this input"recommendation-service-0.0.1-SNAPSHOT.jar" 
or jar should be in some special format?

Reply:
Yes, it is a special format, for example: /deployments/tmp/src/gs-spring-boot-docker-0.1.0.jar
https://medium.com/@plswiderski/deploy-spring-boot-application-to-openshift-3-next-gen-2b311f55f0c5